File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/MemberStatus.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum MemberStatus {
    Active,
    Invited;

    public String toWire() {
        return switch (this) {
            case Active -> "active";
            case Invited -> "invited";
        };
    }

    public static MemberStatus fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "active" -> Active;
            case "invited" -> Invited;
            default -> throw new IllegalArgumentException("Unknown member status: " + value);
        };
    }
}

File: apps/backend2/src/Common/Domain/ValueObjects/MemberStatus.cs
namespace OffceOs.Domain.Common.ValueObjects;

public enum MemberStatus
{
    Active,
    Invited,
}

public static class MemberStatusExtensions
{
    public static string ToStorageString(this MemberStatus status) => status switch
    {
        MemberStatus.Active => "active",
        MemberStatus.Invited => "invited",
        _ => throw new ArgumentOutOfRangeException(nameof(status)),
    };

    public static MemberStatus ToMemberStatus(this string value) => value switch
    {
        "active" => MemberStatus.Active,
        "invited" => MemberStatus.Invited,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown member status: {value}"),
    };
}
